addons/website_livechat/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -0,0 +1,32 @@
+odoo.define('website_livechat/static/src/models/messaging_notification_handler/messaging_notification_handler.js', function (require) {
+'use strict';
+
+const { registerInstancePatchModel } = require('mail/static/src/model/model_core.js');
+
+registerInstancePatchModel('mail.messaging_notification_handler', 'website_livechat/static/src/models/messaging_notification_handler/messaging_notification_handler.js', {
+
+    //----------------------------------------------------------------------
+    // Private
+    //----------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _handleNotificationPartner(data) {
+        const { info } = data;
+        if (info === 'send_chat_request') {
+            this._handleNotificationPartnerChannel(data);
+            const channel = this.env.models['mail.thread'].findFromIdentifyingData({
+                id: data.id,
+                model: 'mail.channel',
+            });
+            this.env.messaging.chatWindowManager.openThread(channel, {
+                makeActive: true,
```

Comment:
Need a js test to cover this if possible.

Apart from that and the other remark, the rest looks good.